sdks/python/apache_beam/options/pipeline_options_validator.py

```diff
@@ -393,3 +402,16 @@ def validate_repeatable_argument_passed_as_list(self, view, arg_name):
       return self._validate_error(
           self.ERR_REPEATABLE_OPTIONS_NOT_SET_AS_LIST, arg, arg_name)
     return []
+
+  # Minimally validates the endpoint url. This is not a strict application
+  # of http://www.faqs.org/rfcs/rfc1738.html.
+  def validate_endpoint_url(self, endpoint_url):
```

Comment:
shall we move this to apache_beam.internal or other util modules since it can be used elsewhere? Then we could just add unit tests independent of PipelineOptions.

Reply:
I would wait until we need it somewhere else.

Reply:
validating URLs is a relatively straightforward thing that we probably don't do often, I see the argument for having it as a util in a more general place but I don't know if it'll get much use